feature_engine/datetime/datetime.py

```diff
@@ -301,6 +342,8 @@ def get_feature_names_out(self, input_features: Optional[List] = None) -> List:
             variables in the transformed dataset (original + new variables) is returned.
             Alternatively, only the names for the datetime features derived from
             input_features will be returned.
+            If the transformer was fitted on the dataframe index, you may only pass
+            `index` as input_features, or leave it as `None`.
```

Comment:
Could we reprhase these 2 sentences as "If the features were derived from the dataframe index, pass 'index' to obtain the datetime features or None for all output features."